build.rs

```diff
@@ -1,39 +1,46 @@
-extern crate gcc;
+extern crate cc;
 
 use std::env;
 
 fn main() {
     let target = env::var("TARGET").unwrap();
-    let mut base_config = gcc::Build::new();
-    base_config.cpp(true)
+    let mut base_config = cc::Build::new();
+    base_config
+        .cpp(true)
         .include("bitcoin/src")
         .include("bitcoin/src/secp256k1/include")
-        .flag("-g").flag("-O2").flag("-std=c++11").flag("-Wno-unused-parameter")
+        .opt_level(2)
+        .debug(true)
+        .flag_if_supported("-std=c++11")
+        .extra_warnings(true)
         .define("__STDC_FORMAT_MACROS", None);
     if target == "x86_64-unknown-linux-gnu" {
-	base_config
-        .define("HAVE_ENDIAN_H", Some("1"))
-        .define("HAVE_DECL_HTOBE16", Some("1"))
-        .define("HAVE_DECL_HTOLE16", Some("1"))
-        .define("HAVE_DECL_BE16TOH", Some("1"))
-        .define("HAVE_DECL_LE16TOH", Some("1"))
-        .define("HAVE_DECL_HTOBE32", Some("1"))
-        .define("HAVE_DECL_HTOLE32", Some("1"))
-        .define("HAVE_DECL_BE32TOH", Some("1"))
-        .define("HAVE_DECL_LE32TOH", Some("1"))
-        .define("HAVE_DECL_HTOBE64", Some("1"))
-        .define("HAVE_DECL_HTOLE64", Some("1"))
-        .define("HAVE_DECL_BE64TOH", Some("1"))
-        .define("HAVE_DECL_LE64TOH", Some("1"));
+        base_config
+            .define("HAVE_ENDIAN_H", Some("1"))
+            .define("HAVE_DECL_HTOBE16", Some("1"))
+            .define("HAVE_DECL_HTOLE16", Some("1"))
+            .define("HAVE_DECL_BE16TOH", Some("1"))
+            .define("HAVE_DECL_LE16TOH", Some("1"))
+            .define("HAVE_DECL_HTOBE32", Some("1"))
+            .define("HAVE_DECL_HTOLE32", Some("1"))
+            .define("HAVE_DECL_BE32TOH", Some("1"))
+            .define("HAVE_DECL_LE32TOH", Some("1"))
+            .define("HAVE_DECL_HTOBE64", Some("1"))
+            .define("HAVE_DECL_HTOLE64", Some("1"))
+            .define("HAVE_DECL_BE64TOH", Some("1"))
+            .define("HAVE_DECL_LE64TOH", Some("1"));
     }
     base_config
         .file("bitcoin/src/utilstrencodings.cpp")
         .file("bitcoin/src/uint256.cpp")
         .file("bitcoin/src/pubkey.cpp")
+        .file("bitcoin/src/hash.cpp")
```

Comment:
Why does windows need additional files?

Reply:
Because `cl.exe` linker is more strict than `gcc`, it doesn't allow unresolved symbols in static libraries.
For example `pubkey.cpp` depends on `BIP32Hash` function from the `hash.cpp`.